pkg/warmup/target.go

```diff
@@ -58,41 +56,41 @@ func NewTarget(readinessHttpClient whttp.Client, readinessGrpcClient grpc.Client
 		options:             options,
 	}
 
-	err := t.waitForReadinessProbe(done)
+	err := t.waitForReadinessProbe()
 	if err != nil {
-		return Target{}, fmt.Errorf("wait for readiness probe: %v", err)
+		return Target{}, fmt.Errorf("Error waiting for target to be ready: %v", err)
 	}
 	return t, err
 }
 
-func (t Target) waitForReadinessProbe(done <-chan struct{}) error {
+func (t Target) waitForReadinessProbe() error {
 
-	log.Printf("waiting for target readiness probe for %ds", t.options.ReadinessTimeoutInSeconds)
+	log.Printf("Waiting for target to be ready for a max of %ds", t.options.ReadinessTimeoutInSeconds)
 
 	timeout := time.After(time.Duration(t.options.ReadinessTimeoutInSeconds) * time.Second)
 	for {
 		select {
 		case <-timeout:
-			return fmt.Errorf("target readiness probe: timeout %d seconds exceeded", t.options.ReadinessTimeoutInSeconds)
-		case <-done:
-			return errors.New("target readiness probe: received done signal")
+			return fmt.Errorf("Giving up! Target not ready after %d seconds 🙁", t.options.ReadinessTimeoutInSeconds)
 		default:
+			// Wait one second between attempts. This is not configurable
+			time.Sleep(time.Second * 1)
```

Comment:
Nice, makes sense.

Do you think this should be even higher? It's similar to K8S probe's [periodSeconds](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#configure-probes) which has a default value of 10. That sounds too much though, maybe makes sense for slow-to-start apps.

Reply:
1 think one is fine. Most apps, including jvm based, start in just a few seconds.